diff -uNr deadbeef-0.5.1.old/plugins/ffmpeg/Makefile.am deadbeef-0.5.1/plugins/ffmpeg/Makefile.am
--- deadbeef-0.5.1.old/plugins/ffmpeg/Makefile.am	2011-03-03 21:06:15.000000000 +0100
+++ deadbeef-0.5.1/plugins/ffmpeg/Makefile.am	2011-06-06 21:19:42.009745288 +0200
@@ -4,6 +4,6 @@
 ffmpeg_la_SOURCES = ffmpeg.c
 ffmpeg_la_LDFLAGS = -module
 
-ffmpeg_la_LIBADD = $(LDADD) $(FFMPEG_DEPS_LIBS)
+ffmpeg_la_LIBADD = $(LDADD) $(FFMPEG_DEPS_LIBS) -lm
 AM_CFLAGS = $(CFLAGS) -std=c99 ${FFMPEG_DEPS_CFLAGS}
 endif
diff --git a/plugins/alac/Makefile.am b/plugins/alac/Makefile.am
index 0dc48ad..268e3e9 100644
--- a/plugins/alac/Makefile.am
+++ b/plugins/alac/Makefile.am
@@ -15,7 +15,7 @@ alac.c decomp.h demux.c demux.h stream.c stream.h\
 
 alac_la_LDFLAGS = -module
 
-alac_la_LIBADD = $(LDADD) $(FAAD2_LIBS)
+alac_la_LIBADD = $(LDADD) $(FAAD2_LIBS) -lm
 AM_CFLAGS = $(CFLAGS) -std=c99 -DUSE_MP4FF -DUSE_TAGGING -I../libmp4ff
 endif
 
